File: Examples/WindowsFormsProject/Models/DataVertex.cs
using GraphX;
using GraphX.PCL.Common.Models;

namespace WindowsFormsProject
{
    /* DataVertex is the data class for the vertices. It contains all custom vertex data specified by the user.
     * This class also must be derived from VertexBase that provides properties and methods mandatory for
     * correct GraphX operations.
     * Some of the useful VertexBase members are:
     *  - ID property that stores unique positive identfication number. Property must be filled by user.
     *  
     */

    public class DataVertex: VertexBase
    {
        /// <summary>
        /// Some string property for example purposes
        /// </summary>
        public string Text { get; set; }
 
        #region Calculated or static props

        public override string ToString()
        {
            return Text;
        }


        #endregion

        /// <summary>
        /// Default parameterless constructor for this class
        /// (required for YAXLib serialization)
        /// </summary>
        public DataVertex():this("")
        {
        }

        public DataVertex(string text = "")
        {
            Text = text;
        }
    }
}

File: Examples/WindowsDesktop_VB.NET_WinForms_Example/Models/DataVertex.vb

Imports GraphX.PCL.Common.Models

Namespace Models

    Public Class DataVertex
        Inherits VertexBase
        ''' <summary>
        ''' Some string property for example purposes
        ''' </summary>
        Public Property Text As String

#Region "Calculated or static props"

        Public Overrides Function ToString() As String
            Return Text
        End Function


#End Region

        ''' <summary>
        ''' Default parameterless constructor for this class
        ''' (required for YAXLib serialization)
        ''' </summary>
        Public Sub New()
            Me.New("")
        End Sub

        Public Sub New(Optional text1 As String = "")
            Text = text1
        End Sub
    End Class
End Namespace